test/e2e/service.go

```diff
@@ -72,9 +72,19 @@ var _ = framework.KubeDescribe("Services", func() {
 	f := framework.NewDefaultFramework("services")
 
 	var c *client.Client
+	var gceController *GCEIngressController
 
 	BeforeEach(func() {
 		c = f.Client
+		By("Initializing gce controller")
+		gceController = &GCEIngressController{ns: f.Namespace.Name, Project: framework.TestContext.CloudConfig.ProjectID, c: c}
+		gceController.init()
+	})
+
+	// Platform specific cleanup
+	AfterEach(func() {
+		By("Cleaning up cloud resources")
+		cleanupGCE(gceController)
```

Comment:
Ah, I got you, thanks.
Sorry for the late response, I got some very critical task recently.


? 2016?9?2??03:36?Prashanth B <notifications@github.com<mailto:notifications@github.com>> ???


In test/e2e/service.go<https://github.com/kubernetes/kubernetes/pull/31862#discussion_r77217362>:

>
>       BeforeEach(func() {
>               c = f.Client
> +             By("Initializing gce controller")
> +             gceController = &GCEIngressController{ns: f.Namespace.Name, Project: framework.TestContext.CloudConfig.ProjectID, c: c}
> +             gceController.init()
> +     })
> +
> +     // Platform specific cleanup
> +     AfterEach(func() {
> +             By("Cleaning up cloud resources")
> +             cleanupGCE(gceController)


you essentially need to do this: https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/delete-stranded-load-balancers.sh#L25

But figure out the expected resource names from the UIDs of the Services created for the particular test, and only delete those resources.

-
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/kubernetes/kubernetes/pull/31862/files/48b7f21e0c74c6e8c46a0ae3fedd2d4a054e9c9f#r77217362>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AP30JJx2hL_7FR5dWPeDTMSY2uMz0Kgqks5qlwtIgaJpZM4JyfRH>.